docs/WIRELESS.md

```diff
@@ -0,0 +1,162 @@
+
+Currently, EVE stays away from managing GSM modem's firmware and expects a modem unit to be fully configured by a hardware manufacturer to provide a [QMI interface](https://en.wikipedia.org/wiki/Qualcomm_MSM_Interface).
+
+### Enabling a new GSM modem (or debugging an existing one)
+
+As with any hardware, you will typically go through the following steps:
+
+1. making sure that the modem device is connected to the host edge node through one of the common busses (this doesn't guarantee that the host can communicate with the device, but at least you should see it registered on the bus)
+2. making sure that the modem is configured in such a way that you can actually communicate with it
+3. making sure that you can drive the modem and instruct it to connect to the wireless provider
+4. making sure that a modem connected to the wireless provider can establish a data connection and reflect it back to the host as a network interface (typically `wwanX`)
+
+#### 1. Basic bus connectivity
+
+Alsmot all well known GSM modems get hooked to a USB bus (most of them communicate using USB2 but some try USB3 and then [trouble may ensue](https://forum.sierrawireless.com/t/mc7455-not-recognized-at-boot/9735) and you may need to [do soldering](https://forum.sierrawireless.com/t/effect-of-removing-usb-3-0-pins-of-mc7455/12702/2)). From there, a single device will expose different interfaces to accommodate various ways of interacting with it. These include:
+
+* traditional serial interface capable of accepting AT-commands (popular choices of drivers include [Qualcom's qcserial](https://github.com/torvalds/linux/blob/master/drivers/usb/serial/qcserial.c) and [Sierra Wireless serial](https://github.com/torvalds/linux/blob/master/drivers/usb/serial/sierra.c)). This interface is NOT capable of directly transmitting Ethernet frames and requires an old-school PPP support
+* traditional serial interface used for special purposes such as [GPS NMEA port](https://en.wikipedia.org/wiki/NMEA_0183) or [Diagnostic Monitoring](https://forum.sierrawireless.com/t/sierra-linux-qmi-dmcapture-sh-not-working/6373)
+* One of the CDC (no, not [that CDC](https://www.cdc.gov/) - rather [communications device class](https://en.wikipedia.org/wiki/USB_communications_device_class)) interfaces capable of direct transport of Ethernet frames:
+  * [CDC MBIM](https://www.kernel.org/doc/Documentation/networking/cdc_mbim.txt)
+  * [CDC QMI](https://en.wikipedia.org/wiki/Qualcomm_MSM_Interface)
+
+To put it all together, a single GSM modem is likely to look something like this when inspected by the `lsusb -t` command:
+
+```bash
```

Comment:
Nit: this might be better formatted as "console" rather than "bash"

Reply:
fixed